### ISSUE TITLE
batchStore(), batchInsert(), etc. with StatementType.STATIC_STATEMENT throws java.sql.BatchUpdateException

### ISSUE THREAD

**Issue by AnghelLeonard** [jOOQ/jOOQ#11509]
### Expected behavior
No exception

### Actual behavior
```text
Caused by: java.sql.BatchUpdateException: The variable name '@result' has already been declared. 
Variable names must be unique within a query batch or stored procedure.
```

### Steps to reproduce the problem
Assume two records:

```Java
SaleRecord sr1 = new SaleRecord();
sr1.setFiscalYear(2004);
sr1.setSale(543.33);
sr1.setEmployeeNumber(1166L);

SaleRecord sr2 = new SaleRecord();
sr2.setFiscalYear(2003);
sr2.setSale(8002.22);
sr2.setEmployeeNumber(1504L);
```
The following batch works fine:

```Java
int[] result = ctx.configuration().derive(new Settings()
   .withStatementType(StatementType.STATIC_STATEMENT)).dsl()
   .batchInsert(sr1)
   .execute();
```

And, the SQL is:

```sql
declare @result table ([sale_id] bigint); insert into [classicmodels].[dbo].[sale] ([fiscal_year], [sale], [employee_number]) output [inserted].[sale_id] into @result values (2004, 543.33, 1166); select [r].[sale_id] from @result [r];
```
But, this one (added `sr2`) throw the ` java.sql.BatchUpdateException` mentioned above:

```Java
int[] result = ctx.configuration().derive(new Settings()
   .withStatementType(StatementType.STATIC_STATEMENT)).dsl()
   .batchInsert(sr1, sr2)
   .execute();
```

The SQL is:

```sql
declare @result table ([sale_id] bigint); insert into [classicmodels].[dbo].[sale] ([fiscal_year], [sale], [employee_number]) output [inserted].[sale_id] into @result values (2004, 543.33, 1166); select [r].[sale_id] from @result [r];

declare @result table ([sale_id] bigint); insert into [classicmodels].[dbo].[sale] ([fiscal_year], [sale], [employee_number]) output [inserted].[sale_id] into @result values (2003, 8002.22, 1504); select [r].[sale_id] from @result [r];
```

### Workaround
As @lukaseder suggested by adding `withRenderOutputForSQLServerReturningClause(Boolean.FALSE)` solve the problem. So, this works:

```Java
int[] result = ctx.configuration().derive(new Settings()
   .withRenderOutputForSQLServerReturningClause(Boolean.FALSE)
   .withStatementType(StatementType.STATIC_STATEMENT))
   .dsl()
   .batchInsert(sr1, sr2)
   .execute();
```

And, the SQL is:

```sql
insert into [classicmodels].[dbo].[sale] ([fiscal_year], [sale], [employee_number]) values (2004, 543.33, 1166)
insert into [classicmodels].[dbo].[sale] ([fiscal_year], [sale], [employee_number]) values (2003, 8002.22, 1504)
```

### Versions

- jOOQ: 3.14
- Java: 14
- Database (include vendor): SQL Server 2014
- OS:-
- JDBC Driver (include name if inofficial driver): Microsoft JDBC

**Comment by lukaseder** [comment]
A quick attempt of fixing this failed with a new exception:

```
org.jooq.exception.DataAccessException: SQL [insert into [test].[dbo].[t_author] ([ID], [LAST_NAME]) output [inserted].[ID] into @alias_3118182 values (9, 'YY'); select [ID] from @alias_3118182 [r];]; Es wurde ein Resultset für den Updatevorgang generiert.
	at org.jooq_3.15.0-SNAPSHOT.SQLSERVER2017.debug(Unknown Source)
	at org.jooq.impl.Tools.translate(Tools.java:2920)
	at org.jooq.impl.DefaultExecuteContext.sqlException(DefaultExecuteContext.java:643)
	at org.jooq.impl.BatchMultiple.execute(BatchMultiple.java:158)
	at org.jooq.impl.BatchMultiple.execute(BatchMultiple.java:93)
	at org.jooq.impl.BatchCRUD.executeStatic(BatchCRUD.java:174)
	at org.jooq.impl.BatchCRUD.execute(BatchCRUD.java:86)
	at org.jooq.test.all.testcases.BatchTests.lambda$1(BatchTests.java:505)
	at org.jooq.test.jOOQAbstractTest.withDSLContext(jOOQAbstractTest.java:1600)
	at org.jooq.test.jOOQAbstractTest.bindAndInline(jOOQAbstractTest.java:1588)
	at org.jooq.test.BaseTest.bindAndInline(BaseTest.java:1633)
	at org.jooq.test.all.testcases.BatchTests.testBatchInsertUpdate(BatchTests.java:482)
	at org.jooq.test.jOOQAbstractTest.testBatchInsertUpdate(jOOQAbstractTest.java:7298)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:93)
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:40)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:529)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:756)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:452)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:210)
Caused by: java.sql.BatchUpdateException: Es wurde ein Resultset für den Updatevorgang generiert.
	at com.microsoft.sqlserver.jdbc.SQLServerStatement.executeBatch(SQLServerStatement.java:1845)
	at org.jooq.tools.jdbc.DefaultStatement.executeBatch(DefaultStatement.java:122)
	at org.jooq.tools.jdbc.DefaultStatement.executeBatch(DefaultStatement.java:122)
	at org.jooq.impl.BatchMultiple.execute(BatchMultiple.java:139)
	... 40 more
```

It seems the underlying problem is that we try to fetch the `OUTPUT` in statement batches, which isn't supported by the mssql-jdbc driver. Supporting this is a popular feature request though, so I'll look at both of these issues at the same time, soon: https://github.com/jOOQ/jOOQ/issues/3327

**Comment by lukaseder** [comment]
Looking into this now. A similar issue appears when trying to write a test for H2 for this problem:

```
org.jooq.exception.DataAccessException: SQL [select "ID" from final table (insert into "PUBLIC"."T_AUTHOR" ("ID", "LAST_NAME") values (9, 'YY')) "T_AUTHOR"]; Methode nicht zulässig für eine Abfrage. Erlaubt sind execute oder executeQuery, nicht jedoch executeUpdate
Method is not allowed for a query. Use execute or executeQuery instead of executeUpdate; SQL statement:
select "ID" from final table (insert into "PUBLIC"."T_AUTHOR" ("ID", "LAST_NAME") values (8, 'XX')) "T_AUTHOR" [90001-210]
	at org.jooq_3.17.0-SNAPSHOT.H2.debug(Unknown Source)
	at org.jooq.impl.Tools.translate(Tools.java:3078)
	at org.jooq.impl.DefaultExecuteContext.sqlException(DefaultExecuteContext.java:670)
	at org.jooq.impl.BatchMultiple.execute(BatchMultiple.java:146)
	at org.jooq.impl.BatchMultiple.execute(BatchMultiple.java:88)
	at org.jooq.impl.BatchCRUD.executeStatic(BatchCRUD.java:174)
	at org.jooq.impl.BatchCRUD.execute(BatchCRUD.java:86)
	at org.jooq.test.all.testcases.BatchTests.lambda$1(BatchTests.java:511)
```

The query is:

```sql
select "ID"
from final table (
  insert into "PUBLIC"."T_AUTHOR" ("ID", "LAST_NAME") 
  values (9, 'YY')
) "T_AUTHOR"
```

The underlying problem is the same. We can't fetch generated keys and other generated values in batch statements. This is a popular feature request as can be seen here: https://github.com/jOOQ/jOOQ/issues/3327

Until that feature request is implemented, jOOQ should turn off returning generated keys whenever batch statements are executed, which would solve this issue here.

**Comment by lukaseder** [comment]
There's an old mechanism implemented in jOOQ 2.4, which omits the `RETURNING` clause in some dialects: https://github.com/jOOQ/jOOQ/issues/1537

That old mechanism should be redundant, I think?

**Comment by lukaseder** [comment]
Yep. No regressions after fixing this and removing #1537

**Comment by lukaseder** [comment]
I'll backport this fix anyway. Given we had #1537, which was a partial fix for some dialects (which somehow didn't apply to SQL Server), I'd say most users don't expect `RETURNING` clauses to be generated.

**Comment by lukaseder** [comment]
Changing issue description as this is explicitly about `batchStore()`, `batchInsert()`, etc., not about other types of batching, where it's the user's responsibility to omit unnecessary `RETURNING` clauses.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.17.0, 3.16.6 (https://github.com/jOOQ/jOOQ/issues/13258), and 3.15.10 (https://github.com/jOOQ/jOOQ/issues/13259).

Thanks again for your report, as always!
